README.md

```diff
@@ -67,12 +67,16 @@ Now you are able to interact with your local, mock version of Amazon Web Service
 You can use [AWS-SDK](http://aws.amazon.com/tools/), or a number of other third-party client tools such as elasticfox.
 
 To manage instances on mock EC2, just point to the custom EC2 endpoint as follows:
-http://localhost:8000/aws-mock/ec2-endpoint/ (equivalent to the official endpoint url https://ec2.us-west-1.amazonaws.com/)
+http://localhost:8001/aws-mock/ec2-endpoint/ (equivalent to the official endpoint url https://ec2.us-west-1.amazonaws.com/)
```

Comment:
Everything builds according to the new way just fine, buI I get an http 404 here:

```
christopherbalz@system76-tlswe-laptop-2:~/code/tlswe/aws-mock$ gradle clean
:clean UP-TO-DATE
:cxf-stub:clean UP-TO-DATE

BUILD SUCCESSFUL

Total time: 2.07 secs
christopherbalz@system76-tlswe-laptop-2:~/code/tlswe/aws-mock$ ./gradlew integrationTomcatRun

Welcome to Gradle 4.7!

Here are the highlights of this release:
 - Incremental annotation processing
 - JDK 10 support
 - Grouped non-interactive console logs
 - Failed tests are re-run first for quicker feedback

For more details see https://docs.gradle.org/4.7/release-notes.html


> Task :compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.6
1 warning
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-jasper/7.0.76/tomcat-embed-jasper-7.0.76.pom
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-logging-juli/7.0.76/tomcat-embed-logging-juli-7.0.76.pom
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-core/7.0.76/tomcat-embed-core-7.0.76.pom
Download http://repo2.maven.org/maven2/org/eclipse/jdt/core/compiler/ecj/4.4.2/ecj-4.4.2.pom
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-el/7.0.76/tomcat-embed-el-7.0.76.pom
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-jasper/7.0.76/tomcat-embed-jasper-7.0.76.jar
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-core/7.0.76/tomcat-embed-core-7.0.76.jar
Download http://repo2.maven.org/maven2/org/eclipse/jdt/core/compiler/ecj/4.4.2/ecj-4.4.2.jar
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-el/7.0.76/tomcat-embed-el-7.0.76.jar
Download http://repo2.maven.org/maven2/org/apache/tomcat/embed/tomcat-embed-logging-juli/7.0.76/tomcat-embed-logging-juli-7.0.76.jar
[2018-12-25 11:09:41,048] [INFO ] [localhost-startStop-1] aws-mock started. (com.tlswe.awsmock.common.listener.AppServletContextListener.java:138)

> Task :integrationTomcatRun
Started Tomcat Server
The Server is running at http://localhost:8001/aws-mock

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 22s
3 actionable tasks: 3 executed
christopherbalz@system76-tlswe-laptop-2:~/code/tlswe/aws-mock$ 
```

Reply:
You need to access `http://localhost:8001/aws-mock/ec2-endpoint/?Action=...` rather than `http://localhost:8001/aws-mock/?Action=...`. 
Btw I've just fixed a NullPointerException (500) there.

Reply:
Ok working for me now - however, there are a couple things:

Could you please update this message so that the user can get a url (so that the message tells the user about a url) that they can use to quickly verify that the basic thing is working:
```
Started Tomcat Server
The Server is running at http://localhost:8001/aws-mock
```
The reason is that ` http://localhost:8001/aws-mock` gives an http 404.

Also, as part of a separate PR, could you make this project conform to our standard `Makefile` based `install`, `start`, `stop`, `clean`, etc. format?  Of course, the user can still use direct `gradle` (and we should give that instruction as an option in the details of the README, in case say a user is on Windows).

Reply:
I've pushed a couple of updates:
- print endpoint urls while starting mock server
- added a Makefile

Reply:
Ok cool, will check soon

Reply:
The server is definitely running (as two `make start`s in a row show), but I still get 404's for all of these:

```
Started Tomcat Server
The Server is running at http://localhost:8001/aws-mock
EC2 endpoint available is at http://localhost:8001/aws-mock/ec2-endpoint
CloudWatch endpoint available is at http://localhost:8001/aws-mock/cloudwatch
```

Maybe there is some config that a default user doesn't have but that you have, which allows these urls to work properly for you?

Also, the doc here: https://github.com/treelogic-swe/aws-mock/wiki/User's-Guide  needs to be updated to port `8001`; it still says `8000`.

Reply:
Are you sure you are getting 404 on either `http://localhost:8001/aws-mock/ec2-endpoint` or `http://localhost:8001/aws-mock/cloudwatch`? 
Even if lacking proper parameters (`Action`, `Version`, etc.) it should be a 400 (bad request) not a 404 (not found). 
Also I've updated the port number to 8001 in the doc.

Reply:
Yes confirmed I am.  If we can't isolate this easily, then how about we make an lxd container instance to check this?  Emek set us up on Fluffy with a full lxd-based test system (provisioned by Ansible) that's on the private GitLab.

Reply:
Great, the curl worked!  

```
christopherbalz@system76-tlswe-laptop-2:~/code/tlswe/aws-mock$ curl "http://localhost:8001/aws-mock/ec2-endpoint/?Version=2016-11-15&Action=DescribeImages"
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:DescribeImagesResponse xsi:type="DescribeImagesResponseType" xmlns="http://ec2.amazonaws.com/doc/2014-02-01/" xmlns:ns2="http://ec2.amazonaws.com/doc/2013-02-01/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <requestId>cf7b9f7c-ebc4-4564-95e4-20677fc0e3da</requestId>
    <imagesSet>
        <item>
            <imageId>ami-00abcdef</imageId>

```

Reply:
Also, at the place where I used to get a 404 ( http://localhost:8001/aws-mock/ec2-endpoint
 ) , I do now get an xml response with the following message:
```
No parameter in query at all! See http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-query-api.html for building a valid query.
```

So, that is pretty good.